cmd/query/app/querysvc/adjuster/otelattribute.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package adjuster
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/pkg/otelsemconv"
+)
+
+var otelLibraryKeys = map[string]struct{}{
+	string(otelsemconv.TelemetrySDKLanguageKey):   {},
+	string(otelsemconv.TelemetrySDKNameKey):       {},
+	string(otelsemconv.TelemetrySDKVersionKey):    {},
+	string(otelsemconv.TelemetryDistroNameKey):    {},
+	string(otelsemconv.TelemetryDistroVersionKey): {},
+}
+
+// OTELAttribute creates an adjuster that removes the OpenTelemetry library attributes
+// from spans and adds them to the attributes of a resource.
+func OTELAttribute() Adjuster {
+	return Func(func(traces ptrace.Traces) (ptrace.Traces, error) {
+		adjuster := otelAttributeAdjuster{}
+		resourceSpans := traces.ResourceSpans()
+		for i := 0; i < resourceSpans.Len(); i++ {
+			rs := resourceSpans.At(i)
+			resource := rs.Resource()
+			scopeSpans := rs.ScopeSpans()
+			for j := 0; j < scopeSpans.Len(); j++ {
+				ss := scopeSpans.At(j)
+				spans := ss.Spans()
+				for k := 0; k < spans.Len(); k++ {
+					span := spans.At(k)
+					adjuster.adjust(span, resource)
```

Comment:
Technically, this is incorrect. You could, in theory, have two spans under the same Resource which have different library attributes, for whatever reason, which would require us to clone the resource. It should never happen in practice if the instrumentation is not messed up.

Reply:
@yurishkuro I see - do we want to fix that?

Reply:
rather than implementing the actual splitting of Resource I would go with soft validation - before copying attribute to Resource check if it already exists and if the value is the same. If the value is different then don't copy and instead add a warning to the span.